dspace-server-webapp/src/test/java/org/dspace/app/rest/PrimaryBitstreamControllerIT.java

```diff
@@ -0,0 +1,296 @@
+import org.dspace.app.rest.test.AbstractControllerIntegrationTest;
+import org.dspace.builder.BitstreamBuilder;
+import org.dspace.builder.BundleBuilder;
+import org.dspace.builder.CollectionBuilder;
+import org.dspace.builder.CommunityBuilder;
+import org.dspace.builder.ItemBuilder;
+import org.dspace.content.Bitstream;
+import org.dspace.content.Bundle;
+import org.dspace.content.Collection;
+import org.dspace.content.Community;
+import org.dspace.content.Item;
+import org.dspace.content.service.BitstreamService;
+import org.dspace.content.service.BundleService;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+
+/**
+ * Tests for the PrimaryBitstreamController
+ */
+public class PrimaryBitstreamControllerIT extends AbstractControllerIntegrationTest {
+
+    @Autowired
+    BundleService bundleService;
+    @Autowired
+    BitstreamService bitstreamService;
+
+    Item item;
+    Bitstream bitstream;
+    Bundle bundle;
+
+    @Before
+    @Override
+    public void setUp() throws Exception {
+        super.setUp();
+
+        context.turnOffAuthorisationSystem();
+        Community community = CommunityBuilder.createCommunity(context).build();
+        Collection collection = CollectionBuilder.createCollection(context, community).build();
+        item = ItemBuilder.createItem(context, collection).build();
+
+        // create bitstream in ORIGINAL bundle of item
+        String bitstreamContent = "TEST CONTENT";
+        try (InputStream is = IOUtils.toInputStream(bitstreamContent, CharEncoding.UTF_8)) {
+             bitstream = BitstreamBuilder.createBitstream(context, item, is)
+                                         .withName("Bitstream")
+                                         .withMimeType("text/plain")
+                                         .build();
+        }
+        bundle = item.getBundles("ORIGINAL").get(0);
+        context.restoreAuthSystemState();
+    }
+
+    @Test
+    public void testGetPrimaryBitstream() throws Exception {
+        bundle.setPrimaryBitstreamID(bitstream);
+
+        String token = getAuthToken(admin.getEmail(), password);
+        getClient(token).perform(get(getBundleUrl(bundle.getID())))
+                        .andExpect(status().isOk())
+                        .andExpect(jsonPath("$", BitstreamMatcher.matchProperties(bitstream)));
+    }
+
+    @Test
+    public void testGetPrimaryBitstreamBundleNotFound() throws Exception {
+        String token = getAuthToken(admin.getEmail(), password);
+        getClient(token).perform(get(getBundleUrl(UUID.randomUUID())))
+                        .andExpect(status().isNotFound());
+    }
+
+    @Test
+    public void testGetPrimaryBitstreamNonExisting() throws Exception {
+        String token = getAuthToken(admin.getEmail(), password);
+        getClient(token).perform(get(getBundleUrl(bundle.getID())))
+                        .andExpect(status().isOk())
```

Comment:
this should be 201 No Content

Reply:
@abollini We've updated that to be 204 No content as I suspect the 201 was a typo.